text/numeral-notation.md

```diff
@@ -0,0 +1,124 @@
+
+  * Z' is now defined like Z, but it is not a definition of integers; it is a definition of string of decimal digits; it is necessary for programming issues, not for mathematical issues; its definition may change if we decide to be able to parse and print decimal numbers (e.g. 3.5), or even rings in general (e.g. polynomials), as it was suggested in the Coq working group;
+
+  * If we move the definition of Z into Datatypes.v, do we move also their associated comments explaining its implementation? Do we have to move all Z functions? Or kept them where they are now? In that case, Z definitions, functions, and proofs would be separated in different parts of the Coq library;
+
+  * Z is very important, Z' is a detail; they should not be considered as the same thing
+
+ * It was suggested to define Numeral Notation for R differently, since the way integer numbers were coded for R was not good; I think that compatibility is important; if R users want reals integer numbers be parsed and printed differently, they can use their own "Numeral Notation" vernac command (in that case, it masks the "Numeral Notation" of the library), test if it is really better and later, we could adopt a new implementation of real integers.
+
+ * The system have been implemented also for int31, bigN, bigZ and bigQ; actually, I think that they should eventually been excluded from this system, because for these types, there are very important issues for speed and memory usage; in this new implementation, there is a big difference (I tested for 10^1000) and it it 4 times slower than the previous version and I suspect this could be worse for bigger numbers.
+
+ * Speed of parsing and printing for the other types (nat, Z, R) should be tested, but it depends whether Coq programmers use really big numbers of these types; if not, we could have this version which is probably slower that the previous one, but small compared to the other parts of Coq (e.g. tactics).
+
+ * Pierre Letouzey : I suggest to use a very simple base-10 encoding as Coq entry point of Daniel's new parsing/printing mechanism, and let Coq do the rest of the conversion between these base-10 numbers and nat/positive/N/Z/. A [proof-of-concept](https://github.com/letouzey/baseconv) is on github.
+
+  * Interest:
+
+   * Less uncertified critical code : during parsing at the OCaml level, we could keep numeral constants as strings (e.g. "123"), and simply convert these strings to Coq lists of base-10 digits, i.e. D1::D2::D3::nil in my first datatype proposal, see [Deci.v](https://github.com/letouzey/baseconv/blob/master/Deci.v), or (D1 (D2 (D3 Stop))) in the second datatype proposal in [DeciBis.v](https://github.com/letouzey/baseconv/blob/master/DeciBis.v).
```

Comment:
How do you propose determining what is a numeral and what is not?  I believe the current code does not change the parsing of numerals, only the interpretation.  When I modified the code to strip off leading zeros for backwards compatibility, there was a noticeable performance impact when I did the stripping in too many places; this code is already somewhat performance sensitive.

If we're not allowing the parsing of strings of anything other than decimal digits, then allowing `list ascii ` seems like useless generalizing that will make the user do more work.